src/ucm/util/replace.c

```diff
@@ -23,8 +23,20 @@
 
 #define MAP_FAILED ((void*)-1)
 
-pthread_mutex_t ucm_reloc_get_orig_lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
-pthread_t volatile ucm_reloc_get_orig_thread = -1;
+pthread_mutex_t ucm_reloc_get_orig_lock = PTHREAD_MUTEX_INITIALIZER;
+
+__attribute__((constructor(101)))
+void ucm_reloc_get_orig_lock_initializer()
+{
+    pthread_mutexattr_t mutattr = {0};
+    pthread_mutexattr_init(&mutattr);
+    pthread_mutexattr_settype(&mutattr, PTHREAD_MUTEX_RECURSIVE);
+    pthread_mutex_init(&ucm_reloc_get_orig_lock, &mutattr);
+    pthread_mutexattr_destroy(&mutattr);
+}
+
+
+pthread_t volatile ucm_reloc_get_orig_thread = (pthread_t)-1;
```

Comment:
maybe add some #define for ((pthread_t)-1) ?

Reply:
Where would you like it?